packages/auth-types/index.d.ts

```diff
@@ -0,0 +1,281 @@
+  verifyPhoneNumber(
+    phoneNumber: string,
+    applicationVerifier: ApplicationVerifier
+  ): Promise<any>;
+}
+
+export class RecaptchaVerifier extends RecaptchaVerifier_Instance {}
+export class RecaptchaVerifier_Instance implements ApplicationVerifier {
+  constructor(
+    container: any | string,
+    parameters?: Object | null,
+    app?: FirebaseApp | null
+  );
+  clear(): any;
+  render(): Promise<any>;
+  type: string;
+  verify(): Promise<any>;
+}
+
+export class TwitterAuthProvider extends TwitterAuthProvider_Instance {
+  static PROVIDER_ID: string;
+  static credential(token: string, secret: string): AuthCredential;
+}
+export class TwitterAuthProvider_Instance implements AuthProvider {
+  providerId: string;
+  setCustomParameters(customOAuthParameters: Object): AuthProvider;
+}
+
+export type UserCredential = {
+  additionalUserInfo?: AdditionalUserInfo | null;
+  credential: AuthCredential | null;
+  operationType?: string | null;
+  user: User | null;
+};
+
+export interface UserMetadata {
+  creationTime?: string;
+  lastSignInTime?: string;
+}
+
+export type Persistence = string;
+
+export class FirebaseAuth {
+  private constructor();
+
+  static Persistence: {
+    LOCAL: Persistence;
+    NONE: Persistence;
+    SESSION: Persistence;
+  };
+
+  app: FirebaseApp;
+  applyActionCode(code: string): Promise<any>;
+  checkActionCode(code: string): Promise<any>;
+  confirmPasswordReset(code: string, newPassword: string): Promise<any>;
+  createUserWithEmailAndPassword(email: string, password: string): Promise<any>;
+  currentUser: User | null;
+  fetchProvidersForEmail(email: string): Promise<any>;
+  getRedirectResult(): Promise<any>;
+  languageCode: string | null;
+  onAuthStateChanged(
+    nextOrObserver: Observer<any, any> | ((a: User | null) => any),
+    error?: (a: Error) => any,
+    completed?: Unsubscribe
+  ): Unsubscribe;
+  onIdTokenChanged(
+    nextOrObserver: Observer<any, any> | ((a: User | null) => any),
+    error?: (a: Error) => any,
+    completed?: Unsubscribe
+  ): Unsubscribe;
+  sendPasswordResetEmail(
+    email: string,
+    actionCodeSettings?: ActionCodeSettings | null
+  ): Promise<any>;
+  setPersistence(persistence: Persistence): Promise<any>;
+  signInAndRetrieveDataWithCredential(credential: AuthCredential): Promise<any>;
+  signInAnonymously(): Promise<any>;
+  signInWithCredential(credential: AuthCredential): Promise<any>;
+  signInWithCustomToken(token: string): Promise<any>;
+  signInWithEmailAndPassword(email: string, password: string): Promise<any>;
+  signInWithPhoneNumber(
+    phoneNumber: string,
+    applicationVerifier: ApplicationVerifier
+  ): Promise<any>;
+  signInWithPopup(provider: AuthProvider): Promise<any>;
+  signInWithRedirect(provider: AuthProvider): Promise<any>;
+  signOut(): Promise<any>;
+  useDeviceLanguage(): any;
+  verifyPasswordResetCode(code: string): Promise<any>;
+}
+
+declare module '@firebase/app-types' {
+  interface FirebaseNamespace {
```

Comment:
I notice this list (and the current typings in prod) is not up to date. There are missing classes:
https://github.com/firebase/firebase-js-sdk/blob/master/packages/firebase/externs/firebase-auth-externs.js#L1944
https://github.com/firebase/firebase-js-sdk/blob/master/packages/firebase/externs/firebase-auth-externs.js#L86

Reply:
I have updated the typings and added the classes mentioned, PTAL.